Admin validation

yuvalelarat opened the pull request:
Added validateAdmin middleware on "/api/validate-admin"

yuvalelarat committed:
[feat]: admin validation middleware

added validation to check if current user's role is admin (0)

yuvalelarat committed:
[feat]: admin validation middleware

yuvalelarat commented:
Please notice me if its good enough

tomered commented on src/index.ts:
```diff
@@ -16,6 +17,7 @@ app.use(express.json());
 app.use("/login", loginRouter);
 app.use("/register", registerRouter);
 app.use("/api", authenticateUser);
+app.use("/api/validate-admin", authenticateUser, validateAdmin);
```
line 19 should look like this:
app.use("/api", authenticateUser, authenticateUser, validateAdmin, adminEndpoint);
delete line 20 and create adminEndpoint

tomered replied:
line 19 should look like this:
app.use("/api", authenticateUser, authenticateUser, validateAdmin);

tomered commented on src/middleware/validateAdmin.ts:
```diff
@@ -0,0 +1,48 @@
+import { verifyJwt } from "../utils/jwt";
+import { JwtPayload } from "jsonwebtoken";
+import { User } from "../entities/user";
+
```
please add documentation and comments

tomered commented on src/middleware/validateAdmin.ts:
```diff
@@ -0,0 +1,48 @@
+    console.error(err);
+    return res.status(500).send("Internal server error.");
+  }
+};
```
add next to middleware function

yuvalelarat committed:
added adminEndpoint

yuvalelarat commented:
I have added adminEndpoint.ts in routes folder and changed to app.use("/api", authenticateUser, validateAdmin, adminEndpoint); + added comments
is that what you meant?

yuvalelarat committed:
fixed

yuvalelarat commented:
done.

yuvalelarat committed:
added documentation
